### ISSUE TITLE
bug: lots of empty strings and collections in yaml editors

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#1785]
When JS objects get serialized as JSON, there isn't anything equivalent to Go's `omitempty` option and that leads to seeing a lot of empty strings, collections, etc. in all our YAML editors.

While this may not be harmful, Kubernetes users are definitely not accustomed to seeing resources represented this way.

Example:

<img width="972" alt="Screenshot 2024-04-06 at 1 08 21 AM" src="https://github.com/akuity/kargo/assets/1821014/28a23d9a-8e87-4341-ae67-f68bc8a755d1">

I am wondering if running objects through something like this before stringifying might be a good idea?

```typescript
function cleanObject(obj: any) {
  for (let key in obj) {
    if (obj[key] === "" || obj[key] === null || obj[key] === undefined || (Array.isArray(obj[key]) && obj[key].length === 0)) {
      delete obj[key];
    } else if (typeof obj[key] === 'object' && obj[key] !== null) {
      cleanObject(obj[key]);
    }
  }
  return obj;
}
```

**Comment by webstradev** [comment]
There is an easier way to do this than cleaning the object @krancour .

JSON.stringify can take a replacer function as an argument and then you can set anything you want to undefined which should hide it from the stringify result.

**Comment by krancour** [comment]
Thanks @webstradev! I am admittedly not a front-end guy. I dabble if it's a tiny change, but otherwise put my trust in @rpelczar and @rbreeze. 👍 on this suggestion!

**Comment by krancour** [comment]
Since most languages don't have anything comparable to Go's `omitempty`, the agreement we came to internally was that we would add an option (new request field) to applicable endpoints, for the benefit of non-Go client, to return YAML or JSON as bytes or a string in a new response field.

I believe this will mainly be applicable to _get_ endpoints, so that the back-end can return the JSON or YAML of a single object in preparation for display as a "live manifest" or in preparation for editing in a YAML editor.

* AnalysisRun
* AnalysisTemplate
* Freight
* Promotion
* Stage
* Warehouse

**Comment by hiddeco** [comment]
This got addressed for AnalysisRuns in #1863. But as discovered there (see https://github.com/akuity/kargo/pull/1863#issuecomment-2064491693), the other modals make use of the returned list response at present to display the data.

To solve the issue for these, they need to start using the get endpoints instead.
